src/manifests/input_manifest.py

```diff
@@ -49,6 +54,9 @@ class InputManifest(ComponentManifest):
                 "name": {"required": True, "type": "string"},
                 "version": {"required": True, "type": "string"},
                 "patches": {"type": "list", "schema": {"type": "string"}},
+                "platform": {"type": "string"},
+                "architecture": {"type": "string"},
+                "snapshot": {"type": "boolean"},
```

Comment:
Rather than make these features of the build, these seem like part the CI section, maybe as command line args?

Reply:
It's actually a feature of the build in the build manifest, so I think this is desirable. The fact that they are passed command line is not material IMO.